### PR TITLE
Fix for issue #5209 - Consolidated SafeAccessTokenHandle and SafeTokenHandle

### PR DESCRIPTION
**Pull request by AlexBHarley**
Deleted SafeAccessTokenHandle and replaced occurences with SafeTokenHandle

### CODE REVIEW

File: src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs [hunk 1]
```diff
@@ -45,6 +43,14 @@ internal void InitialSetHandle(IntPtr h)
             base.handle = h;
```
Comment (Clockwork-Muse) [inline src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs]: Is this (`InitialSetHandle`) even used?  Given that the handle will be valid after the constructor returns, I'm not sure we need it.

File: src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs [hunk 2]
```diff
@@ -19,15 +19,13 @@
 namespace Microsoft.Win32.SafeHandles
 {
     [SecurityCritical]
-    internal sealed class SafeTokenHandle : SafeHandle
+    public sealed class SafeTokenHandle : SafeHandle
     {
         private const int DefaultInvalidHandleValue = 0;
```
Comment (Clockwork-Muse) [inline src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs]: You left this definition, but `InvalidHandle` uses `IntPtr.Zero`.  You should probably pick one or the other (I feel like `IntPtr.Zero` reads a little cleaner).

File: src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs [hunk 3]
```diff
@@ -19,15 +19,13 @@
...
 
-        internal static readonly SafeTokenHandle InvalidHandle = new SafeTokenHandle(new IntPtr(DefaultInvalidHandleValue));
+        private SafeTokenHandle() : base(new IntPtr(DefaultInvalidHandleValue), true) { }
```
Comment (Clockwork-Muse) [inline src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs]: No clue if stuff like this gets inlined, but you could delegate to `this(IntPtr.Zero)` (or creating the new pointer).

File: src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs [hunk 4]
```diff
@@ -45,6 +43,14 @@ internal void InitialSetHandle(IntPtr h)
             base.handle = h;
         }
 
+        public static SafeTokenHandle InvalidHandle
+        {
+            get
+            {
+                return new SafeTokenHandle(IntPtr.Zero);
```
Comment (Clockwork-Muse) [inline src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs]: Given the chain of constructors, you could use the default constructor here.

**Comment by stephentoub** [comment]
@AlexBHarley, thank you for your work on this.  However, if we're going to consolidate these two types, we need to conslidate in the other direction.  SafeAccessTokenHandle is public (both in corefx and the full framework) whereas SafeTokenHandle is internal and an implementation detail... if we're going to get rid of one, it needs to be getting rid of SafeTokenHandle, not SafeAccessTokenHandle.

**Comment by AlexBHarley** [comment]
@stephentoub so I am moving SafeAccessTokenHandle into common? And removing duplicates of the SafeTokenHandle class from System.Security.AccessControl and System.Diagnostics.Process?

**Comment by stephentoub** [comment]
@AlexBHarley, no.  

This particular issue is just about the copy of SafeTokenHandle that's in System.Security.Principal.Windows (though I see that comments on the issue mention other things).  The SafeTokenHandle that's in System.Security.Principal.Windows appears to actually be dead code... the fix should be as simple as deleting https://github.com/dotnet/corefx/blob/master/src/System.Security.Principal.Windows/src/Microsoft/Win32/SafeHandles/SafeTokenHandle.cs and removing the reference to it from System.Security.Principal.Windows.csproj.

Then there's the issue of two other SafeTokenHandle.cs files used in System.Diagnostics.Process and System.Security.AccessControl.  As a separate change, those two files could be consolidated into one in Common, with those projects changed to use the shared one rather than each having their own copy.

**Comment by stephentoub** [comment]
@AlexBHarley, thanks for your interest in contributing here.  Since this change would need to be redone anyway before it could be merged, I'm going to go ahead and close this (I don't want it to be merged accidentally).  Please feel free to submit a new PR if you want to tackle the issue as I outlined above. Thanks!

**Pull request closed without merging**